src/deserialize.zig

```diff
@@ -183,7 +184,9 @@ pub fn deserialize(comptime T: type, allocator: Allocator, serialized: []const u
             if (serialized.len == 0 or serialized[0] == rlpByteListShortHeader or serialized[0] == rlpListLongHeader) {
                 out.* = null;
                 // 0 if serialized was empty, one in the case of an empty list
-                return serialized.len;
```

Comment:
This line was incorrect. `serialized` is the full remaining bytes to process, not only the bytes scoped to the optional. 

This was found by trying to deserialize a tx sent to the zero address for creating a contract.